op-dispute-mon/mon/forecast.go

```diff
@@ -139,6 +142,9 @@ func (f *Forecast) forecastGame(game *monTypes.EnrichedGameData, metrics *foreca
 				"rootClaim", game.RootClaim, "expected", expected)
 		} else {
 			metrics.AgreeDefenderAhead++
+			if metrics.LatestValidL2Block < game.L2BlockNumber {
+				metrics.LatestValidL2Block = game.L2BlockNumber
+			}
```

Comment:
I suspect we should update this for any game we agree with, regardless of who is currently forecast to win.  If we agree, it means that in our view the proposal is valid. We expect the game to eventually resolve as `DefenderWins` but its not unusual for the challenger to be ahead at different points (and there are metrics that allow alerting if the challenger stays ahead too long which indicates op-challenger isn't doing its job properly).

Reply:
yes I'd just filter on agreement, so the metric reports when the last valid proposal was made and doesn't care about whether or not the game resolved correctly (there's plenty of metrics to flag if a game is going to resolve incorrectly already).